non-critical-infra/modules/vaultwarden.nix

```diff
@@ -0,0 +1,114 @@
+{ config, ... }:
+{
+  services.vaultwarden = {
+    enable = true;
+    backupDir = "/var/lib/bitwarden_rs/backups/";
+    environmentFile = "/var/lib/bitwarden_rs/vaultwarden.env";
+    config = {
+      DOMAIN = "https://vault.nixos.org";
+      SIGNUPS_ALLOWED = false;
+      SHOW_PASSWORD_HINT = false;
+      ROCKET_ADDRESS = "127.0.0.1";
+      ROCKET_PORT = 8222;
+      ROCKET_LOG = "critical";
+      SMTP_HOST = "mail.gandi.net";
+      SMTP_PORT = 587;
+      SMTP_SSL = true;
+      SMTP_FROM = "vaultwarden@nixos.org";
+      SMTP_FROM_NAME = "NixOS Vaultwarden";
+    };
+  };
+
+  services.nginx = {
+    enable = true;
+    recommendedGzipSettings = true;
+    recommendedOptimisation = true;
+    recommendedProxySettings = true;
+    recommendedTlsSettings = true;
+    virtualHosts."vault.nixos.org" = {
+      forceSSL = true;
+      enableACME = true;
+      locations."/" = {
+        proxyPass = "http://127.0.0.1:${toString config.services.vaultwarden.config.ROCKET_PORT}";
+      };
+    };
+  };
+
+  sops.secrets = {
+    vaultwarden-env = {
+      sopsFile = ../secrets/vaultwarden-env.caliban;
+      format = "binary";
+      path = "/var/lib/bitwarden_rs/vaultwarden.env";
+    };
+
+    vaultwarden-backup-pass = {
+      sopsFile = ../secrets/vaultwarden-backup-pass.caliban;
+      format = "binary";
+    };
+  };
+
+  services.borgbackup.jobs.vaultwarden = {
+    encryption = {
+      passCommand = "cat /run/secrets/vaultwarden-backup-pass";
+      mode = "repokey-blake2";
+    };
+    repo = "/mnt/backup-share/caliban.nixos.org/vaultwarden";
```

Comment:
borgbackup can backup directly to a storagebox also. We don't need to mount the cifs here. Or are there other reasons we want it mounted?